packages/contextual-help/src/ContextualHelp.ts

```diff
@@ -0,0 +1,205 @@
+            return;
+        }
+
+        import('@spectrum-web-components/popover/sp-popover.js');
+
+        const template = html`
+            <sp-popover class="popover">
+                <section>
+                    ${this.headline &&
+                    html`
+                        <h2 class="heading">${this.headline}</h2>
+                    `}
+                    <slot></slot>
+                    <div class="link">
+                        <slot name="link"></slot>
+                    </div>
+                </section>
+            </sp-popover>
+        `;
+
+        render(template, event.target as HTMLElement);
+    }
+
+    private renderDialog(event: SlottableRequestEvent): void {
+        event.stopPropagation();
+
+        if (event.data === removeSlottableRequest) {
+            render(undefined, event.target as HTMLElement);
+            return;
+        }
+
+        import('@spectrum-web-components/dialog/sp-dialog-wrapper.js');
+        const headlineVisibility = !this.headline ? 'none' : undefined;
+
+        const template = html`
+            <sp-dialog-wrapper
+                dismissable
+                underlay
+                headline=${ifDefined(this.headline)}
+                headline-visibility=${ifDefined(headlineVisibility)}
+            >
+                <slot></slot>
+                <div class="link">
+                    <slot name="link"></slot>
+                </div>
+            </sp-dialog-wrapper>
+        `;
+
+        render(template, event.target as HTMLElement);
+    }
+
+    protected override render(): TemplateResult {
+        const actualPlacement = this.isMobile.matches
+            ? undefined
+            : this.placement;
+
+        return html`
+            <sp-action-button
+                quiet
+                size="s"
+                id="trigger"
+                aria-label=${this.buttonAriaLabel}
+                ?is-active=${this.overlayOpen}
+            >
+                ${this.variant === 'help'
+                    ? html`
+                          <sp-icon-help-outline
+                              slot="icon"
+                          ></sp-icon-help-outline>
+                      `
+                    : html`
+                          <sp-icon-info-outline
+                              slot="icon"
+                          ></sp-icon-info-outline>
+                      `}
+            </sp-action-button>
+            <sp-overlay
+                trigger="trigger@click"
+                placement=${ifDefined(actualPlacement)}
+                type=${this.isMobile.matches ? 'modal' : 'auto'}
+                receives-focus="true"
+                .offset=${this.offset}
+                @sp-opened=${() => {
+                    this.overlayOpen = true;
+                }}
+                @sp-closed=${() => {
+                    this.overlayOpen = false;
+                }}
+                @slottable-request=${this.isMobile.matches
```

Comment:
Do you want to also offer the option for a consumer to lazily render the content of this overlay?

Reply:
This does not seem like the kind of component you interact with way too many times, as its content does not change often.
So I am thinking it would be a good practice to lazy load its content. On the other hand, its content is indeed modest. What is your opinion on this?

Reply:
I think we'd want it at some point. It is a performance optimization, so I'm OK with excluding it for the initial release, but prefer to take your lead as you're the one actually doing the work.

Reply:
I'd ship it as it is for a component which _theoretically_ should have just some text inside.